pkg/matcher/annotation_matcher.go

```
@@ -508,3 +516,38 @@ func MatchRunningPipelineRunForIncomingWebhook(eventType, incomingPipelineRun st
 	}
 	return nil
 }
+
+// resolveCustomParamsForCEL resolves custom parameters from the Repository CR for use in CEL expressions.
+// It returns a map of parameter names to values, excluding reserved keywords.
+// All parameters are returned as strings, including those from secret_ref.
+func resolveCustomParamsForCEL(ctx context.Context, repo *apipac.Repository, event *info.Event, cs *params.Run, vcx provider.Interface, eventEmitter *events.EventEmitter, logger *zap.SugaredLogger) map[string]string {
+	if repo == nil || repo.Spec.Params == nil {
+		return map[string]string{}
+	}
+
+	// Create kubeinteraction interface
+	kinteract, err := kubeinteraction.NewKubernetesInteraction(cs)
+	if err != nil {
+		logger.Warnf("failed to create kubernetes interaction for custom params: %s", err.Error())
+		return map[string]string{}
```

If there is one non existing secrets all other customparams are going to be skipped, should we just continue instead ?

---

Continuing here would mask configuration issues (like non-existent or misconfigured secrets) and make debugging harder. If we return partial customparams, users might see inconsistent cel evaluation results without understanding why some params work and others don't.
Returning an error forces users to fix misconfigurations upfront and keeps them informed. [`CustomParams.GetParams`](https://github.com/openshift-pipelines/pipelines-as-code/blob/main/pkg/customparams/customparams.go#L65) does return partial results with error but IMO failing completely makes debugging easier than silently using incomplete data